src/main/scala/ie/zalando/fabric/gateway/service/IngressDerivationChain.scala

```diff
@@ -90,8 +90,33 @@ class IngressDerivationChain(stackSetOperations: StackSetOperations)(implicit ma
       val finalRoutes = gateway.corsConfig.fold(skipperRoutes) { corsConfig =>
         withCors(gateway, meta.name, corsConfig, skipperRoutes)
       }
-      combineBackendsAndRoutes(backends, finalRoutes, meta)
+      val ingressDefinitions = combineBackendsAndRoutes(backends, finalRoutes, meta)
+
+      gateway.serviceProvider match {
+        case StackSetProvidedServices(_, _) if versionedHostsEnabled => appendServiceSpecificHosts(ingressDefinitions)
+        case _ => ingressDefinitions
+      }
+    }
+  }
+
+  def appendServiceSpecificHosts(ingresses: List[IngressDefinition]): List[IngressDefinition] = {
+    val serviceSpecificIngresses = for {
+      ingress <- ingresses
+      hostMapping <- ingress.hostMappings
+      splitHost = hostMapping.host.split('.')
+      if splitHost.length == 4 && splitHost(2) == "zalan" && splitHost(3) == "do"
```

Comment:
Just thought about this logic again and think it's bad, since it's an open source project. I think instead I can take the desired base url as an env variable and set the host to `svc.$baseUrl`